dask/dataframe/core.py

```diff
@@ -304,6 +304,19 @@ def size(self):
         return self.reduction(methods.size, np.sum, token='size', meta=int,
                               split_every=False)
 
+    @property
+    def shape(self):
+        """
+        Return a tuple representing the dimensionality of the DataFrame.
+        """
+        from dask.delayed import delayed
+        if isinstance(self._meta, pd.DataFrame):
+            col_size = len(self.columns)
+            row_size = delayed(int)(self.size / col_size)
+            return delayed(tuple)((row_size, col_size))
+        else:
+            return delayed(tuple)((self.size,))
```

Comment:
Given the type selection in here perhaps we should define this separately on the DataFrame and Series classes directly?

Reply:
Ok, I will separate it